### PR TITLE
Merging Register and Update operations into one

### PR DESCRIPTION
**Pull request by qiangdavidliu**
Merging Register and Update operations into one for registry, registrationChannel and replicationChannel. Calling this new operation Register for now.

Also removed 3 unused test-utils classes.

### CODE REVIEW

File: eureka-client/src/main/java/com/netflix/eureka2/client/channel/RegistrationChannelImpl.java [hunk 1]
```diff
@@ -43,17 +40,17 @@ public RegistrationChannelImpl(TransportClient transportClient, RegistrationChan
 
     @Override
     public Observable<Void> register(final InstanceInfo instanceInfo) {
-        if (!moveToState(STATES.Idle, STATES.Registered)) {// State check. Only register if the state is Idle.
+        if (!moveToState(STATES.Idle, STATES.Registered) &&
+            !moveToState(STATES.Registered, STATES.Registered)) {
```
Comment (tbak) [inline eureka-client/src/main/java/com/netflix/eureka2/client/channel/RegistrationChannelImpl.java]: We can replace moveToState(STATES.Registered, STATES.Registered) with state.get() == STATES.Registered.

File: eureka-client/src/test/java/com/netflix/eureka2/client/registration/RegistrationHandlerImplTest.java [hunk 2]
```diff
@@ -64,17 +63,10 @@ public void testRegistersClientsOverDifferentChannel() throws Exception {
     @Test
     public void testConvertsRegistrationToUpdateIfAlreadyRegistered() throws Exception {
```
Comment (tbak) [inline eureka-client/src/test/java/com/netflix/eureka2/client/registration/RegistrationHandlerImplTest.java]: Better test name? We do not convert to updates, but test multiple registrations for the same service.

File: eureka-write-server/src/main/java/com/netflix/eureka2/server/channel/ReceiverReplicationChannel.java [hunk 3]
```diff
@@ -231,14 +195,16 @@ public void call() {
...
 
         return registry.unregister(toUnregister, replicationSource)
-                .ignoreElements()
-                .cast(Void.class)
-                .doOnCompleted(new Action0() {
+                .map(new Func1<Boolean, Void>() {
                     @Override
-                    public void call() {
+                    public Void call(Boolean aBoolean) {
+                        // an emit means the tempNewInfo was successfully registered
```
Comment (tbak) [inline eureka-write-server/src/main/java/com/netflix/eureka2/server/channel/ReceiverReplicationChannel.java]: Comment copy-paste. Does not fit here.

File: eureka-write-server/src/main/java/com/netflix/eureka2/server/channel/RegistrationChannelImpl.java [hunk 4]
```diff
@@ -139,54 +132,13 @@ public void onNext(Boolean status) {
...
-    }
-
-    @Override
     public Observable<Void> unregister() {
+        logger.debug("Unregistering service in registry: {}", currentInfo);
+
         if (!moveToState(STATES.Registered, STATES.Closed)) {
             STATES currentState = state.get();
             if (currentState == STATES.Idle) {
+                sendErrorOnTransport(INSTANCE_ALREADY_REGISTERED_EXCEPTION);
```
Comment (tbak) [inline eureka-write-server/src/main/java/com/netflix/eureka2/server/channel/RegistrationChannelImpl.java]: Invalid error.

Comment (qiangdavidliu) [inline eureka-write-server/src/main/java/com/netflix/eureka2/server/channel/RegistrationChannelImpl.java]: Will fix this

**Pull request merged**